styx-api-service/src/main/java/com/spotify/styx/api/WorkflowResource.java

```diff
@@ -77,12 +92,80 @@ public WorkflowResource(Storage storage) {
             rc -> patchState(arg("cid", rc), arg("wfid", rc), rc.request())),
         Route.with(
             json(), "PATCH", BASE + "/<cid>/state",
-            rc -> patchState(arg("cid", rc), rc.request()))
+            rc -> patchState(arg("cid", rc), rc.request())),
+        Route.with(
+            json(), "POST", BASE + "/<cid>",
+            rc -> createOrUpdateWorkflow(arg("cid", rc), rc)
+        ),
+        Route.with(
+            json(), "DELETE", BASE + "/<cid>/<wfid>",
+            rc -> deleteWorkflow(arg("cid", rc), arg("wfid", rc), rc)
+        )
     );
 
     return Api.prefixRoutes(routes, V2, V3);
   }
 
+  private Response<ByteString> deleteWorkflow(String cid, String wfid, RequestContext rc) {
+    Request request = Request.forUri(schedulerApiUrl("workflows", cid, wfid), "DELETE");
+    try {
+      return rc.requestScopedClient().send(request).toCompletableFuture().get();
```

Comment:
Better not to block explicitly in user code, instead letting Apollo handle it. You can find some examples in `BackfillResource`.